x-pack/plugins/maps/public/classes/layers/new_vector_layer_wizard/wizard.tsx

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Component, Fragment } from 'react';
+import { EuiEmptyPrompt, EuiPanel, EuiCallOut } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { createNewIndexAndPattern } from './create_new_index_pattern';
+import { RenderWizardArguments } from '../layer_wizard_registry';
+import { VectorLayer } from '../vector_layer';
+import { ESSearchSource } from '../../sources/es_search_source';
+import { ADD_LAYER_STEP_ID } from '../../../connected_components/add_layer_panel/view';
+import { getIndexNameFormComponent } from '../../../kibana_services';
+
+interface State {
+  indexName: string;
+  indexNameError: string;
+  indexingTriggered: boolean;
+  createIndexError: string;
+}
+
+export class NewVectorLayerEditor extends Component<RenderWizardArguments, State> {
+  private _isMounted: boolean = false;
+
+  state: State = {
+    indexName: '',
+    indexNameError: '',
+    indexingTriggered: false,
+    createIndexError: '',
+  };
+
+  componentDidMount() {
+    this._isMounted = true;
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  async componentDidUpdate() {
+    if (this.props.currentStepId === ADD_LAYER_STEP_ID && !this.state.indexingTriggered) {
+      this.setState({ indexingTriggered: true });
+      await this._createNewIndex();
+    }
+  }
+
+  _setCreateIndexError(errorMessage: string) {
+    if (!this._isMounted) {
+      return;
+    }
+    this.setState({
+      createIndexError: errorMessage,
+    });
+  }
+
+  _createNewIndex = async () => {
+    let indexPatternId: string | undefined;
+    try {
+      const response = await createNewIndexAndPattern(this.state.indexName);
+      indexPatternId = response.indexPatternId;
```

Comment:
response is of type CreateDocSourceResp which does not include indexPatternId. In fact, createNewIndexAndPattern only creates the index pattern. You will need another step to create the index pattern. See https://github.com/elastic/kibana/blob/master/x-pack/plugins/file_upload/public/components/json_upload_and_parse.tsx#L180

Also, don't you index the drawn shapes here as well?

Reply:
discussed offline. this is the https://github.com/elastic/kibana/blob/f238ccbdc34a07b7fee44423fb38eebf6f8d637c/x-pack/plugins/maps/common/types.ts#L8

Will keep as is. It creates an index and index-pattern in a single step. It does not add data. Adding data (which requires enabling "edit mode" for a layer) is being worked on here https://github.com/elastic/kibana/pull/99812